providers/provider_data.go

```diff
@@ -44,3 +44,27 @@ func (p *ProviderData) GetClientSecret() (clientSecret string, err error) {
 	}
 	return string(fileClientSecret), nil
 }
+
+func (p *ProviderData) setProviderDefaults(name string, defaultLoginURL, defaultRedeemURL, defaultProfileURL, defaultValidateURL *url.URL, defaultScope string) {
+	p.ProviderName = name
+	setURLDefault(&p.LoginURL, defaultLoginURL)
+	setURLDefault(&p.RedeemURL, defaultRedeemURL)
+	setURLDefault(&p.ProfileURL, defaultProfileURL)
+	setURLDefault(&p.ValidateURL, defaultValidateURL)
+
+	if p.Scope == "" {
+		p.Scope = defaultScope
+	}
+}
+
+// setURLDefault will set the url to the default given if it is empty or nil.
+// If the provided default is nil, it will populate it with an empty URL to make sure it is not nil.
```

Comment:
This seems a bit awkward. Would the double pointer be better solved by allowing us to assume they are setup correctly in a `func NewProviderData() *ProviderData` function? Also not really sure why these are pointers to start with.

Reply:
Yeah it was a bit awkward, I had another go and just let it return the default value, does this look any better?